tests/foundry/test/StoreTest.t.sol

```diff
@@ -12,22 +12,30 @@ contract Store {
 
 contract StoreTest is Test {
 
-    Store myStore;
-
-    function setUp() public {
+    function testAccesses() public {
+        Store myStore = new Store();
         vm.record();
-        myStore = new Store();
-    }
 
-    function testAccesses() public {
         (bytes32[] memory reads, bytes32[] memory writes) = vm.accesses(address(myStore));
         assertEq(reads.length, 1);
         assertEq(writes.length, 1);
     }
 
     function testStoreLoad() public {
+        Store myStore = new Store();
         vm.store(address(myStore), bytes32(uint256(0)), bytes32(uint256(31337)));
         bytes32 testNumber = vm.load(address(myStore), bytes32(uint256(0)));
         assertEq(uint256(testNumber), 31337); // 31337
     }
+
+    function testStoreLoadNonExistent() public {
+        vm.store(address(0), bytes32(uint256(3)), bytes32(uint256(31337)));
+        bytes32 testNumber = vm.load(address(0), bytes32(uint256(3)));
+        assertEq(uint256(testNumber), 31337); // 31337
+    }
+
+    function testLoadNonExistent() public {
+        bytes32 testNumber = vm.load(address(100), bytes32(uint256(23)));
+        assertEq(uint256(testNumber), 0);
+    }
```

Comment:
If you make a copy of this test, that then does:

- Transfer to the address(100).
- Asserts the remaining gas is equal to some value (amke sure it passes on Foundry).

Then we'll have higher confidence in the correctness of this implementation.

Reply:
You could assert the gas amount before and after the call.

Reply:
Working on it. After a first test I was a bit surprised, because the gas used by the KEVM was not just a little bit off but completely off. Digging into this, the reason is that we currently override the rule for `CALL` to implement cheatcodes, and at this point, the gas for the whole call has already been deducted. Under normal execution, `CALL` would refund any unused gas, but our cheatcode rules don't do that, which results in the big difference in gas usage between Foundry and KEVM.

I believe the solution is to implement the cheatcodes by matching against `#exec [ CALL ]` instead of `CALL`. Normally, `#exec [ CALL ]` rewrites to `#gas [CALL, ...] CALL ...`, but when implementing cheatacodes we could simply remove the `#gas` from the rewrite in order to not deduct any gas. Looking at the Rust code I think this would be consistent with Foundry, but I'll double-check.

Reply:
Okay, load/store should now use the same amount of gas under both Foundry and KEVM. It wasn't necessary to match against `#exec [ CALL ]`, we just need to refund all the reserved gas via `#refund`. Also, `#loadAccount` now inserts the loaded account into the access set. (There is also `<touchedAccounts>`, but it doesn't seem to be used inside the semantics, so I'm not sure if it also needs to be updated.)

To make the gas tests pass I had to slightly modify the initial configuration. In particular, I had to replace some symbolic variables with concrete values.

Reply:
Nice :)